testunit.go

```diff
@@ -10,102 +9,88 @@ import (
 	"github.com/Sirupsen/logrus"
 	"github.com/huawei-openlab/oct/factory"
 	"github.com/huawei-openlab/oct/utils"
-	"github.com/huawei-openlab/oct/utils/hooks"
+	"github.com/huawei-openlab/oct/utils/config"
 )
 
 const TestCacheDir = "./bundles/"
 
-func validate(validateObj string, configArgs string) error {
-	generateConfigs(validateObj, configArgs)
-	prepareBundle(validateObj)
+type TestUnit struct {
+	//Case name
+	Name string
+	//Args is used to generate bundle
+	Args string
+	//Describle what does this unit test for. It is optional.
+	Description string
 
-	myruntime, err := factory.CreateRuntime(Runtime)
-	if err != nil {
-		logrus.Printf("Create runtime %v err: %v\n", Runtime, err)
-	}
-	if myruntime.NeedConvert() {
-		if _, err := myruntime.Convert(validateObj, TestCacheDir); err != nil {
-			logrus.Fatalln(err)
-		}
-	}
+	BundleDir string
+	Runtime   factory.Factory
+}
 
-	testopt := utils.GetAfterNStr(configArgs, "--args=./runtimetest --args=", 3)
-	testRoot := path.Join(TestCacheDir, validateObj)
-	out, err := myruntime.StartRT(testRoot)
-	if err != nil {
-		//logrus.Printf("Run runtime err: %v\n", err)
-		return err
+func LoadTestUnits(filename string) (units []TestUnit) {
+	for key, value := range config.BundleMap {
+		//TODO: config.BundleMap should support 'Description'
+		unit := TestUnitNew(key, value, "")
+		units = append(units, unit)
 	}
-	switch testopt {
-	case "vna":
-		err = hooks.SetPostStartHooks(out, hooks.NamespacePostStart)
-	default:
-	}
-
-	return nil
+	return units
 }
 
-func generateConfigs(validateObj string, configArgs string) {
-	args := splitArgs(configArgs)
+func TestUnitNew(name string, args string, desc string) (unit TestUnit) {
+	unit.Name = name
+	unit.Args = args
+	unit.Description = desc
 
-	logrus.Debugf("Args to the ocitools generate: ")
-	for _, a := range args {
-		logrus.Debugln(a)
-	}
-	Mutex.Lock()
```

Comment:
It was not deleted, seems 'git diff' is not readable.
You can read the original code:
https://github.com/liangchenye/oct/blob/master/testunit.go